src/main/java/com/google/sps/DataServlet.java

```diff
@@ -171,53 +180,99 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.setHeader("serverError", e.getMessage());
       return;
     }
-
-    List<String> queried = new ArrayList<>();
-    if (nodeNameParam.length() > 0) {
-      queried.add(nodeNameParam);
-    }
-    // If the token is contained, then get the nodes associated with the token and
-    // add them to the queried nodes
+    // Show mutations relevant to nodes that contain the token in the new graph
     if (currDataGraph.tokenMap().containsKey(tokenParam)) {
       queried.addAll(currDataGraph.tokenMap().get(tokenParam));
     }
+
+    // Truncate the graph from the nodes that the client had searched for
     truncatedGraph = currDataGraph.getReachableNodes(queried, depthNumber);
+    // OPT: check queried and currDataGraph.tokenMap().get(tokenParam) are the same
+    MutableGraph<GraphNode> truncatedGraphNext =
+        currDataGraph.getReachableNodes(currDataGraph.tokenMap().get(tokenParam), depthNumber);
```

Comment:
Do you think that if node name parameter is non-empty it should be one of the "roots" as well?

Reply:
yea i think i made that change in the next PR -- should I combine them together??

Reply:
No I think it's fine to just merge it at that point :)